specs/core/0_beacon-chain.md

```diff
@@ -500,16 +497,29 @@ The following data structures are defined as [SimpleSerialize (SSZ)](https://git
 }
 ```
 
-#### `ProposalSignedData`
+#### `Proposal`
 
 ```python
 {
     # Slot number
     'slot': 'uint64',
     # Shard number (`BEACON_CHAIN_SHARD_NUMBER` for beacon chain)
     'shard': 'uint64',
-    # Block's hash of root
+    # Block root
     'block_root': 'bytes32',
+    # Signature
+    'signature': 'bytes96',
+}
+```
```

Comment:
Why do we even need to do this? Why not be more radical and just remove the Proposal class entirely and directly sign the block? Or is that to reduce the number of slashing conditions down the line when we have shard blocks?

Reply:
We need the `Proposal` class for attester slashings. Maybe there's a way to clean this up.

Reply:
I do like the idea of having two separate slashing conditions for two reasons:

1) It's somewhat cleaner (don't need `Proposal` and `BEACON_CHAIN_SHARD_NUMBER`)
2) Beacon chain proposer slashing is arguably more important than shard proposer slashing. So with two separate transaction types we have guaranteed throughput for beacon chain proposer slashings.

Reply:
It's not completely trivial if we remove `Proposal`. The reason is that for the proposer slashing you would need a Merkle path to the block root to prove the slot number.

Reply:
Wait, why? Why not just have a `ProposerSlashing` object that contains two block headers? Sure it's maybe a hundred bytes longer, but that's nothing compared to attester slashings in any case...

Reply:
We could replace `Proposal` with a `BeaconBlockHeader` object but would that be more verbose (`Proposal` only have 4 fields, `BeaconBlockHeader` would have 7) and slightly less efficient?

Reply:
Aah I see, `BeaconBlock` objects themselves are too big, and we _technically_ don't have a `BeaconBlockHeader` (well, from any block you can create a block header with an equivalent SSZ root by replacing the body with a hash of the body, but if we're going to take advantage of this fact in the spec it would be good to document it well...

I see a few possibilities:

1. Modify the `BeaconBlock` object so it has an explicit header that contains `hash(body)` and a separate body
2. _Define_ a `BeaconBlockHeader` as a header that can be derived from any `BeaconBlock`
3. Keep the current approach

Note that in terms of hashes and bytes, (1) and (2) are exactly equivalent; object1 containing the SSZ root of object2 with object2 living "somewhere else", and obect1 literally containing object2 in its entirety, are in some sense exactly the same thing and will have exactly the same SSZ root. So it's a choice of presentation rather than content.

(3) would change things less relative to the status quo and it would not introduce this weirdness of the header being kinda part of and kinda separate from the rest of the block, though it would lead to a slightly longer spec. I guess my "don't change things at this stage of the game unless it's a really really good idea" bias does make me lean slightly toward (3)....

Reply:
Ok, I'll keep as is and mull over it a bit.

Reply:
Can you approve the PR?